lsp-rust.el

```diff
@@ -587,6 +581,111 @@ The command should include `--message=format=json` or similar option."
 
 (defvar-local lsp-rust-analyzer-inlay-hints-timer nil)
 
+(defface lsp-rust-analyzer-inlay-face
+  '((t :inherit font-lock-comment-face))
+  "The face to use for the Rust Analyzer inlays."
+  :group 'lsp-rust
+  :package-version '(lsp-mode . "7.0"))
+
+(defface lsp-rust-analyzer-inlay-type-face
+  '((t :inherit lsp-rust-analyzer-inlay-face))
+  "Face for inlay type hints (e.g. inferred types)."
+  :group 'lsp-rust)
+
+(defcustom lsp-rust-analyzer-inlay-type-space-prefix nil
```

Comment:
Can't `-space-prefix` and `-prefix` be merged together? Same thing for `-posfix` too.
Merging them still provide same functionality though.

And it avoids providing multiple customization that do the same thing.

Reply:
Alternatively, to save on number of customize options, we can even provide something like `..-inlay-type-format`, that's just a string to pass to `(format lsp-rust-analyzer-inlay-type-format type)`.
For example, set the format to `": %s"` and we will have what it looks like right now.
Just an alternative suggestion though.

Reply:
`-space-prefix` and `-prefix` don't do the same thing, as explained in their documentation strings, one is part of the hint (and uses the same face as the hint) and the other uses the default face. In the screenshot, the chain inlay hints have a space before them, then an arrow and a space inside of them.

I like the idea of the format string, but I want to have two of them for each kind of inlay, one for the inlay text and one for the spacing around the inlay text.

Would that be satisfactory to everybody?

Reply:
> I like the idea of the format string, but I want to have two of them for each kind of inlay, one for the inlay text and one for the spacing around the inlay text.

Great. If you want to use that, then the format can be changed to something like `" %s %s"`, with the first phrase is `-prefix`, and the second phrase is actual hint. If you propertize the string before pass it to `format` it will retain format for each phrase

Reply:
I'm not following, how would that work for having both prefixes and postfixes? Am I missing something? Could you please provide an example that handles all 4 cases (space-prefix, prefix, postfix, space-postfix) as shown in the screenshot [here](https://github.com/emacs-lsp/lsp-mode/pull/2005#issuecomment-670585154)?

Reply:
Ah, sorry, I kinda forgot my previous suggestion :smile:.
You're right, if you want to maintain different face for `-space-prefix` and `-prefix`, it will require two formats instead of one.